### PR TITLE
StorableObject refactor (Cherry picked)

### PR DESCRIPTION
**Pull request by madhavajay**
## Description
The aim of this PR is to simplify things around StorableObject, including StorableObject, send, get, SaveObjectAction, GetObjectResponseMessage, maybe more.

Cherry Picked this: https://github.com/OpenMined/PySyft/pull/5104

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
@xutongye This is a really awesome job cleaning up the whole ugly Wrapper system.

File: src/syft/core/store/storeable_object.py [hunk 1]
```diff
@@ -147,27 +165,13 @@ def _proto2object(proto: StorableObject_PB) -> Serializable:
...
-        # need a different way to get obj_type
-        if proto.obj_type.endswith("ProtobufWrapper"):
-            module_parts = proto.obj_type.split(".")
-            klass = module_parts.pop().replace("ProtobufWrapper", "")
-            proto_type = getattr(sys.modules[".".join(module_parts)], klass)
-            obj_type = proto_type.serializable_wrapper_type
+        # # TODO: FIX THIS SECURITY BUG!!! WE CANNOT USE
+        # #  PYDOC.LOCATE!!!
+        # # Step 2: get the type of wrapper to use to deserialize
+        data_type = pydoc.locate(proto.data_type)
```
Comment (tudorcebere) [inline src/syft/core/store/storeable_object.py:171]: Ideally here we should be able to use the AST to solve the types, the main problem being that we don't have the wrapper types in our AST. Do you think we can find a workaround on pydoc.locate (maybe add the wrapper types in the AST/in some kind of cache). Not a must for this PR, but it's something worth to think on.

File: src/syft/lib/python/bool.py [hunk 2]
```diff
@@ -266,45 +265,14 @@ def _proto2object(proto: Bool_PB) -> "Bool":
...
-        description: Optional[str],
-        tags: Optional[List[str]],
-    ) -> StorableObject:
-        setattr(data, "_id", id)
-        data.tags = tags
-        data.description = description
-        return data
-
-
-aggressive_set_attr(obj=Bool, name="serializable_wrapper_type", attr=BoolWrapper)
```
Comment (tudorcebere) [inline src/syft/lib/python/bool.py:310]: :fire: :firecracker:

**Review by tudorcebere — APPROVED** [review]
The main things left around would be:
* add wrappers to a cache/even the AST.
* remove pydoc.locate from the codebase
* write a few tests on the new wrapping mechanisms to be 100% sure that it's acting as we expect.

This PR is quite a chunky boy, I'll approve it and you can continue your work in a new PR.

Amazing work, one of my all time fav PRs, so much dead/useless code removed.

File: src/syft/ast/klass.py [hunk 3]
```diff
@@ -355,55 +359,16 @@ def send(
...
-            obj=outer_self.object_ref, name="proto", attr=Serializable.proto
-        )
-        to_bytes_attr = "to_bytes"
-        # int has a to_bytes already, so we can use _to_bytes internally
-        if hasattr(outer_self.object_ref, to_bytes_attr):
-            to_bytes_attr = "_to_bytes"
-        aggressive_set_attr(
-            obj=outer_self.object_ref, name=to_bytes_attr, attr=Serializable.to_bytes
-        )
-
```
[the remark this answers is not in the document]
Comment (xutongye) [reply src/syft/ast/klass.py:406]: This still exists, I just move it to another place, `generate_wrapper.py`  :)

File: src/syft/core/store/storeable_object.py (hunk 1 above)
Comment (xutongye) [reply src/syft/core/store/storeable_object.py:171]: I think we can use function `index_syft_by_module_name`. I already add wrapper classes to AST path. But not really add them, just fake it like it's there. Please take a look inside `index_syft_by_module_name`. For exmaple, torch.deviceWrapper has a ast path syft.wrappers.torch.deviceWrapper

**Pull request closed without merging**